### PR TITLE
[New Rule] Execution via local SxS Shared Module

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/423

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by Samirbous** [comment]
Example of Alert:

image

File: rules/windows/execution_shared_modules_local_sxs_dll.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+creation_date = "2020/10/28"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/10/28"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies creation, change or deletion of a DLL module within a Windows SxS local folder. Adversaries may abuse shared modules to
+execute malicious payloads by instructing the Windows module loader to load DLLs from arbitrary local paths.
```
Comment (bm11100) [inline rules/windows/execution_shared_modules_local_sxs_dll.toml:11]:
```suggestion
Identifies the creation, change, or deletion of a DLL module within a Windows SxS local folder. Adversaries may abuse shared modules to
execute malicious payloads by instructing the Windows module loader to load DLLs from arbitrary local paths.
```

File: rules/windows/execution_shared_modules_local_sxs_dll.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Execution via local SxS Shared Module"
+note = "SxS DotLocal folder is a legit feature that can be abused to hijack standard modules loading order by forcing an executable on the same application.exe.local folder to load a malicious DLL module from the same directory. "
```
Comment (bm11100) [inline rules/windows/execution_shared_modules_local_sxs_dll.toml:18]:
```suggestion
note = "The SxS DotLocal folder is a legitimate feature that can be abused to hijack standard modules loading order by forcing an executable on the same application.exe.local folder to load a malicious DLL module from the same directory."
```

**Review by bm11100 — APPROVED** [review]
minor suggestions, otherwise looks good

File: rules/windows/execution_shared_modules_local_sxs_dll.toml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+note = "The SxS DotLocal folder is a legitimate feature that can be abused to hijack standard modules loading order by forcing an executable on the same application.exe.local folder to load a malicious DLL module from the same directory."
+references = ["https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-redirection"]
+risk_score = 43
+rule_id = "a3ea12f3-0d4e-4667-8b44-4230c63f3c75"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Execution"]
+type = "eql"
+
+query = '''
+file where wildcard(file.path, "C:\\*\\*.exe.local\\*.dll")
```
Comment (brokensound77) [inline rules/windows/execution_shared_modules_local_sxs_dll.toml:27]:
```suggestion
file where file.path : "C:\\*\\*.exe.local\\*.dll"
```

**Review by brokensound77 — APPROVED** [review]
just the one syntax change, then LGTM 👍

File: rules/windows/execution_shared_modules_local_sxs_dll.toml (hunk 3 above)
Comment (rw-access) [reply rules/windows/execution_shared_modules_local_sxs_dll.toml]: can you add `file.extension : "dll"`, too?

**Pull request merged**